proxy/http2/Http2Frame.h

```diff
@@ -0,0 +1,252 @@
+/** @file
+
+  Http2Frame
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include "P_Net.h"
+
+#include "HTTP2.h"
+
+/**
+   Incoming HTTP/2 Frame
+ */
+class Http2Frame
+{
+public:
+  Http2Frame(const Http2FrameHeader &h, IOBufferReader *r, bool e = false) : _hdr(h), _ioreader(r), _from_early_data(e) {}
+
+  // Accessor
+  IOBufferReader *reader() const;
+  const Http2FrameHeader &header() const;
+  bool is_from_early_data() const;
+
+private:
+  Http2FrameHeader _hdr;
+  IOBufferReader *_ioreader = nullptr;
+  bool _from_early_data     = false;
+};
+
+/**
+   Outgoing HTTP/2 Frame
+ */
+class Http2TxFrame
+{
+public:
+  Http2TxFrame(const Http2FrameHeader &h) : _hdr(h) {}
+
+  // Don't allocate on heap
+  void *operator new(std::size_t)   = delete;
+  void *operator new[](std::size_t) = delete;
+
+  virtual int64_t xmit(MIOBuffer *iobuffer) const = 0;
```

Comment:
The name is confusing because the implementation doesn't transmit data. I'd name it `write_to`.

Reply:
Ah, this is not new, but still, because you are cleaning up the interface.

Reply:
Right, the name comes from the old interface. Actually, I renamed this `write_to` once and restore it to minimize change. Well, I'll rename again.